pallets/subtensor/src/subnets/weights.rs

```diff
@@ -452,50 +527,29 @@ impl<T: Config> Pallet<T> {
         uids.len() <= subnetwork_n as usize
     }
 
-    #[allow(clippy::arithmetic_side_effects)]
-    pub fn can_commit(netuid: u16, who: &T::AccountId) -> bool {
-        if let Some((_hash, commit_block)) = WeightCommits::<T>::get(netuid, who) {
-            let interval: u64 = Self::get_commit_reveal_weights_interval(netuid);
-            if interval == 0 {
-                return true; //prevent division by 0
-            }
-
-            let current_block: u64 = Self::get_current_block_as_u64();
-            let interval_start: u64 = current_block.saturating_sub(current_block % interval);
-            let last_commit_interval_start: u64 =
-                commit_block.saturating_sub(commit_block % interval);
-
-            // Allow commit if we're within the interval bounds
-            if current_block <= interval_start.saturating_add(interval)
-                && interval_start > last_commit_interval_start
-            {
-                return true;
-            }
+    pub fn is_reveal_block_range(netuid: u16, commit_block: u64) -> bool {
+        let current_block: u64 = Self::get_current_block_as_u64();
+        let commit_epoch: u64 = Self::get_epoch_index(netuid, commit_block);
+        let current_epoch: u64 = Self::get_epoch_index(netuid, current_block);
 
-            false
-        } else {
-            true
-        }
+        // Reveal is allowed only in the epoch immediately after the commit's epoch
```

Comment:
it should be only allowed for one epoch, in the epoch N epochs after the commit epoch, where N might be equal to 1 by default or it might be a higher natural number, as configured by a hyperparameter